src/frequenz/channels/_generic.py

```diff
@@ -0,0 +1,24 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Generic type variables."""
+
+from typing import TypeVar
+
+ChannelMessageT = TypeVar("ChannelMessageT")
+"""The type of the message that can be sent across a channel."""
+
+ErroredChannelT_co = TypeVar("ErroredChannelT_co", covariant=True)
+"""The type of channel having an error."""
+
+MappedMessageT_co = TypeVar("MappedMessageT_co", covariant=True)
+"""The type of the message received by the receiver after being mapped."""
+
+ReceiverMessageT_co = TypeVar("ReceiverMessageT_co", covariant=True)
+"""The type of the message received by a receiver."""
+
+SenderMessageT_co = TypeVar("SenderMessageT_co", covariant=True)
+"""The type of the message sent by a sender."""
+
+SenderMessageT_contra = TypeVar("SenderMessageT_contra", contravariant=True)
+"""The type of the message sent by a sender."""
```

Comment:
Should the docstring be a bit different here to highlight the subtle difference with  `SenderMessageT_co`? Otherwise the docstrings for `SenderMessageT_co` and `SenderMessageT_contra` is the same

Reply:
I don't think so, this is an implementation detail the user doesn't have to care about. The user only needs to know that type is the type of message sent by the sender.

Also the suffixes `_co` and `_contra` are some sort of standard in Python, they are very well known (`pylint` even check for them) and the documentation still shows the full definition too:

![image](https://github.com/frequenz-floss/frequenz-channels-python/assets/1031485/6cee3112-d8e6-4d27-a2c3-5b5c4f1445a7)